engine/schema/src/com/cloud/storage/dao/StoragePoolTagsDaoImpl.java

```diff
@@ -77,4 +90,71 @@ public void deleteTags(long poolId) {
         txn.commit();
     }
 
+    @Override
+    public List<StoragePoolTagVO> searchByIds(Long... stIds) {
+        String batchCfg = _configDao.getValue("detail.batch.query.size");
+
+        final int detailsBatchSize = batchCfg != null ? Integer.parseInt(batchCfg) : 2000;
+
+        // query details by batches
+        List<StoragePoolTagVO> uvList = new ArrayList<StoragePoolTagVO>();
+        int curr_index = 0;
+
+        if (stIds.length > detailsBatchSize) {
+            while ((curr_index + detailsBatchSize) <= stIds.length) {
+                Long[] ids = new Long[detailsBatchSize];
+
+                for (int k = 0, j = curr_index; j < curr_index + detailsBatchSize; j++, k++) {
+                    ids[k] = stIds[j];
+                }
+
+                SearchCriteria<StoragePoolTagVO> sc = StoragePoolIdsSearch.create();
```

Comment:
@nvazquez lines 105-119 are the same as 128-142.
What about extracting them for a method?